arbitrum-docs/for-devs/quickstart-solidity-hardhat.md

```diff
@@ -0,0 +1,491 @@
+| **Data** (cupcakes)   | Stored only in your **browser**. (Usually, stored by centralized infrastructure.)       | Stored on your **device** in an **emulated Ethereum network** (via smart contract).        | Stored on Ethereum's **decentralized test network** (via smart contract).              | Stored on Ethereum's **decentralized mainnet network** (via smart contract).              |
+| **Logic** (vending)   | Served from **Offchain's servers**. Executed by your **browser**.                       | Stored and executed by your **locally emulated Ethereum network** (via smart contract).    | Stored and executed by Arbitrum's **decentralized test network** (via smart contract). | Stored and executed by Arbitrum's **decentralized mainnet network** (via smart contract). |
+| **Presentation** (UI) | Served from **Offchain's servers**. Rendered and executed by your **browser**.          | <- same                                                                                    | <- same                                                                                | <- same                                                                                   |
+| **Money**             | Devs and users pay centralized service providers for server access using fiat currency. | <- same, but only for the presentation-layer concerns (code that supports frontend UI/UX). | <- same, but devs and users pay **testnet $ETH** to testnet validators.                | <- same, but instead of testnet $ETH, they use **mainnet $ETH**.                          |
+
+
+Next, we'll deploy our smart contract to a network of real nodes: Arbitrum's Goerli testnet.
+
+### Deploy the smart contract to the Arbitrum Goerli testnet
+
+We were able to deploy to a local testnet for free because we were using [Hardhat's built-in Ethereum network emulator](https://hardhat.org/hardhat-network/docs/overview#hardhat-network). Because Arbitrum's Goerli testnet is powered by a real network of real nodes, we'll need to pay a small transaction fee to deploy our smart contract. This fee will be paid with the Arbitrum Goerli testnet's token, $AGOR.
+
+:::info $AGOR ISN'T REAL
+
+$AGOR isn't a canonical term, or a real token. It's just convenient shorthand for "Arbitrum Goerli testnet ETH" that we use to make this quickstart more readable.
+
+:::
+
+First, update the `hardhat.config.js` file to specify the private key of the test account that you'll use to deploy your smart contract (and pay the transaction fee):
+
+```javascript title="hardhat.config.js"
+// ...
+const GOERLI_TESTNET_PRIVATE_KEY = ""; // <- this should **not** begin with "0x"
+// ...
+accounts: [GOERLI_TESTNET_PRIVATE_KEY] // <- uncomment this line
+// ...
+```
+
+:::caution
+
+Note that we're adding a private key to a config file. This is **not** a best practice. In a real application, you should use a [signer](https://docs.ethers.io/v5/api/signer/) to sign transactions instead of handling private keys directly.
```

Comment:
I'm not sure this is technically correct. You are already using a signer in the code (to deploy the contract, iirc). But, it's true that devs should use other means to store their keys. We can mention "environment variables" as an example. 
Actually, in a future iteration we could use this method in this quickstart, instead of describing it as an alternative here.